### PR TITLE
API Remove GraphQL

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1799

### CODE REVIEW

File: tests/behat/features/operate-subsites.feature [hunk 1]
```diff
@@ -28,7 +28,7 @@ Feature: Create and select a subsite
     And I press the "Add folder" button
     And I select "Main site" from "SubsitesSelect"
     # Using a short folder name so that it doesn't get truncated on the frontend
-    And I fill in "Folder name" with "mfol"
+    And I fill in "Form_folderCreateForm_Name" with "mfol"
```
Comment (emteknetnz) [inline tests/behat/features/operate-subsites.feature]: For some reason we now need to get the same field via ID rather than name

File: src/Middleware/InitStateMiddleware.php [hunk 2]
```diff
@@ -23,7 +23,6 @@ class InitStateMiddleware implements HTTPMiddleware
      */
     private static $admin_url_paths = [
         'dev/',
-        'graphql/',
```
Comment (NightJar) [inline src/Middleware/InitStateMiddleware.php:26]: You need a hook in whatever consumes this config in order to prevent issues that would block people who _do_ use GraphQL with e.g. custom features in their CMS.

Comment (emteknetnz) [reply src/Middleware/InitStateMiddleware.php:26]: It's private static config, people can update it via their project yml

File: tests/behat/features/operate-subsites.feature [hunk 3]
```diff
@@ -25,10 +25,13 @@ Feature: Create and select a subsite
 
     # Add a file to the main site
     When I go to "admin/assets"
-    And I press the "Add folder" button
     And I select "Main site" from "SubsitesSelect"
+    # Ensure that the new subsite is located before adding folders
+    And I go to "admin/assets"
+    And I press the "Add folder" button
+    And I wait for 2 seconds
```
Comment (ChloeSartorelli) [inline tests/behat/features/operate-subsites.feature]: Do you remember why this change was necessary? Feels like it could be a regression or some underlying bug we're just testing around?

File: tests/behat/features/operate-subsites.feature (hunk 1 above)
Comment (ChloeSartorelli) [reply tests/behat/features/operate-subsites.feature]: Uhhh that feels bad. Did the name of the form field change or something? This feels like it could lead to unexpected regressions, e.g. if someone has js or css relying on form field names?

File: tests/behat/features/operate-subsites.feature [hunk 4]
```diff
@@ -65,10 +68,11 @@ Feature: Create and select a subsite
     Then I should see "file1"
 
     # Add a file to the subsite
-    When I go to "admin/assets"
-    And I select "My subsite" from "SubsitesSelect"
+    When I select "My subsite" from "SubsitesSelect"
+    And I go to "admin/assets"
```
Comment (ChloeSartorelli) [inline tests/behat/features/operate-subsites.feature]: Is this change for the same reason as the one with "Ensure that the new subsite is located before adding folders" above?

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: tests/behat/features/operate-subsites.feature (hunk 3 above)
Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: I cannot remember

File: tests/behat/features/operate-subsites.feature (hunk 4 above)
Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: I think so

File: tests/behat/features/operate-subsites.feature (hunk 1 above)
Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: I don't think it changed, behat was just fussy for some reason

Comment (ChloeSartorelli) [reply tests/behat/features/operate-subsites.feature]: If nothing changed, nothing needs to change here.

But evidently you've had to change something here, which means something _did_ change. Behat being fussy doesn't explain it, because this change isn't necessary in the current `4` branch, it's only needed when applying the changes in these PRs.

So _something_ you've changed has caused this to need to change, and that's scaring me a little that we don't know what it was or what other side effects that might have.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: tests/behat/features/operate-subsites.feature (hunk 1 above)
Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: I'll try changing it back and see if it's resolved, does work locally, we'll see if CI likes it

Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: Works with it reverted on latest kitchen sink run

File: tests/behat/features/operate-subsites.feature (hunk 3 above)
Comment (ChloeSartorelli) [reply tests/behat/features/operate-subsites.feature]: Tried reverting this locally and it passed. Please revert this as it doesn't seem to be necessary.

File: tests/behat/features/operate-subsites.feature (hunk 4 above)
Comment (ChloeSartorelli) [reply tests/behat/features/operate-subsites.feature]: Tried reverting this locally and it passed. Please revert this as it doesn't seem to be necessary (keep the "wait for 2 seconds" below, which does seem to be necessary)

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

**Comment by emteknetnz** [comment]
Have reverted

File: tests/behat/features/operate-subsites.feature [hunk 5]
```diff
@@ -1,4 +1,4 @@
-@javascript
+@javascript @sboyd
```
Comment (ChloeSartorelli) [inline tests/behat/features/operate-subsites.feature]:
```suggestion
@javascript
```

File: tests/behat/features/operate-subsites.feature [hunk 6]
```diff
@@ -25,6 +25,7 @@ Feature: Create and select a subsite
 
     # Add a file to the main site
     When I go to "admin/assets"
+    And I select "Main site" from "SubsitesSelect"
```
Comment (ChloeSartorelli) [inline tests/behat/features/operate-subsites.feature]:
```suggestion
```
That's being done after "add folder" two lines down.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: tests/behat/features/operate-subsites.feature (hunk 5 above)
Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: Removed

File: tests/behat/features/operate-subsites.feature (hunk 6 above)
Comment (emteknetnz) [reply tests/behat/features/operate-subsites.feature]: Removed

**Review by GuySartorelli — APPROVED** [review]
LGTM
